src/debugger/types.js

```diff
@@ -10,21 +10,57 @@
 /* @flow */
 
 export type DebuggerRequest = {
+  id: number,
   command: string,
   arguments: DebuggerRequestArguments,
 };
 
-export type DebuggerRequestArguments = BreakpointArguments | RunArguments;
+export type DebuggerRequestArguments = BreakpointArguments | RunArguments | StackframeArguments;
 
 export type BreakpointArguments = {
-  requestID: number,
   kind: "breakpoint",
   filePath: string,
   line: number,
   column: number,
 };
 
 export type RunArguments = {
-  requestID: number,
   kind: "run",
 };
+
+export type StackframeArguments = {
+  kind: "stackframe",
+};
+
+export type Stackframe = {
+  id: number,
+  fileName: string,
+  line: number,
+  column: number,
+  functionName: string,
+};
+export type DebuggerResponse = ReadyResponse | BreakpointAddResponse | BreakpointStoppedResponse | StackframeResponse;
```

Comment:
To be consistent with DebuggerRequest, define it as:
```
export type DebuggerResponse = {
  id: number,
  kind: string,
  arguments: ....
};
```

Reply:
I can change it to be similar to DebuggerRequest, but the `kind` field needs to go into each argument type to be used for type refinement by flow.